### PR TITLE
Feat(event): Get Function For Object

### PR DESCRIPTION
**Pull request by johndy2742**
### Changes Made

- add get object logic

### Purpose

- add get functionality for Object managament

### How Has This Been Tested

- Postman request test
image

### Related Links

https://chat.digitalservice.id/boards/team/nii5pbyw1idwxrxjd8bp56mtkw/bte55h31wfjgp9xgry174ur3mae/vbyyag4aadi8ejgca6gtqzs55gc

Project: Event Management System
Participants: @johndy2742, @iqbal167, @alqinsidev, @MNAlfiansyah , @ristianaditya

### CODE REVIEW

File: service/src/modules/object/transport/handler/get_objects.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+	page, err := strconv.Atoi(pageStr)
+	if err != nil {
+		http.Error(w, "Invalid page parameter", http.StatusBadRequest)
+		return
+	}
+
+	perPage, err := strconv.Atoi(perPageStr)
+	if err != nil {
+		http.Error(w, "Invalid per_page parameter", http.StatusBadRequest)
```
Comment (iqbal167) [inline service/src/modules/object/transport/handler/get_objects.go:23]: for pagination use, you can see the example here https://github.com/iqbal167/tahura/blob/main/src/modules/tahura/transport/http/request/request.go

and for query implementation in repository in https://github.com/iqbal167/tahura/blob/main/src/modules/tahura/repository/order.go find the `GetOrders` method

Comment (iqbal167) [reply service/src/modules/object/transport/handler/get_objects.go:23]: reference is https://www.stackhawk.com/blog/golang-sql-injection-guide-examples-and-prevention/#:~:text=A%20parametrized%20query%20looks%20much,FROM%20users%20WHERE%20ID%20%3D%20%3F%22

Comment (johndy2742) [reply service/src/modules/object/transport/handler/get_objects.go:23]: thanks for the refference im currently refactoring the code to requirement

File: service/src/modules/object/transport/handler/handler.go [hunk 2]
```diff
@@ -20,3 +23,46 @@ func InitHandler(app *app.App, endpoint endpoint.EndpointInterface) *Handler {
...
+	case _error.ErrEmptyUserName:
+		helperresponse.Render(w, r, h.responseMapping, httpresponse.EmptyUserName, nil, result)
+	case _error.ErrInvalidConfig:
+		helperresponse.Render(w, r, h.responseMapping, httpresponse.InternalServerError, nil, result)
+	case _error.ErrInvalidQRString:
+		helperresponse.Render(w, r, h.responseMapping, httpresponse.InvalidQRString, nil, result)
+	default:
+		helperresponse.Render(w, r, h.responseMapping, httpresponse.BadRequest, nil, nil)
+	}
+}
```
Comment (iqbal167) [inline service/src/modules/object/transport/handler/handler.go:68]: use only what you need to handle error responses

File: service/src/app/logger.go
Comment (iqbal167) [inline service/src/app/logger.go:1]: please remove this file, we will discuss later for logs

File: service/src/modules/object/endpoint/get_objects.go [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+	for i, obj := range objects {
+		responseObj := &response.Object{}
+		if err := copier.Copy(responseObj, obj); err != nil {
+			return nil, err
+		}
+		responseObjects[i] = *responseObj
+	}
+
+	log.Printf("GetObjects Response: %#v", responseObjects)
```
Comment (iqbal167) [inline service/src/modules/object/endpoint/get_objects.go:29]: please remove this log

File: service/src/app/logger.go
Comment (johndy2742) [reply service/src/app/logger.go:1]: ok kang

File: service/src/modules/object/repository/object.go [hunk 4]
```diff
@@ -31,3 +36,124 @@ func (r *Repository) CreateObject(ctx context.Context, obj request.Object) (requ
...
+        organizer,
+        status,
+        created_at,
+        updated_at
+    FROM object
+    WHERE 1 = 1 %s `,
+		r.filterObjectQuery(params, &binds))
+
+	result, err := r.getObjects(ctx, query, binds...)
+	fmt.Println(result)
```
Comment (iqbal167) [inline service/src/modules/object/repository/object.go:60]: remove this too

File: service/src/modules/object/repository/object.go [hunk 5]
```diff
@@ -31,3 +36,124 @@ func (r *Repository) CreateObject(ctx context.Context, obj request.Object) (requ
...
+		println(err)
+		return nil, err
+	}
+	return result, nil
+
+}
+
+func (r *Repository) getObjects(ctx context.Context, query string, args ...interface{}) ([]entity.Object, error) {
+	log.Printf("query :%#v ", query)
+	log.Printf("args :%#v ", args...)
```
Comment (iqbal167) [inline service/src/modules/object/repository/object.go:71]: remove this log

File: service/src/modules/object/transport/handler/get_objects.go [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+	helperresponse "github.com/jabardigitalservice/super-app-services/event/src/helper/http/response"
+	helperutils "github.com/jabardigitalservice/super-app-services/event/src/helper/utils"
+	"github.com/jabardigitalservice/super-app-services/event/src/modules/object/transport/handler/http/request"
+	httpresponse "github.com/jabardigitalservice/super-app-services/event/src/response/http"
+)
+
+func (h *Handler) GetObjects(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	queryParam := request.GetRequestParam(r)
+	log.Printf("PageSize from request: %d", queryParam.PageSize)
```
Comment (iqbal167) [inline service/src/modules/object/transport/handler/get_objects.go:17]: remove this log

**Review by iqbal167 — APPROVED** [review]

**Pull request merged**